src/webgpu/shader/validation/shader_io/builtins.spec.ts

```diff
@@ -109,6 +111,9 @@ g.test('stage_inout')
         !t.params.use_struct,
       'missing @builtin(position) in the vertex output when the vertex output is not a struct'
     );
+    if (t.params.name === 'subgroup_id' || t.params.name === 'num_subgroups') {
```

Comment:
Instead of skipping the test for these cases, fail the test when the builtin is one of these and the feature is not supported.

One way to do this is to extend the table to include an optional field about which language feature (if any) supports the builtin.